src/main/java/seedu/address/logic/parser/SortBuyersCommandParser.java

```diff
@@ -42,7 +48,19 @@ public SortBuyersCommand parse(String args) throws ParseException {
         if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
             Order order = ParserUtil.parseOrder(argMultimap.getValue(PREFIX_NAME).get());
             Comparator<Name> nameComparator = new NameComparator(order);
-            buyerComparator = new BuyerComparator(nameComparator, null);
+            buyerComparator = new BuyerComparator(nameComparator, null, null);
+        }
+
+        if (argMultimap.getValue(PREFIX_PRICE_RANGE).isPresent()) {
+            Order order = ParserUtil.parseOrder(argMultimap.getValue(PREFIX_PRICE_RANGE).get());
+            Comparator<PriceRange> priceRangeComparator = new PriceRangeComparator(order);
+            buyerComparator = new BuyerComparator(null, priceRangeComparator, null);
```

Comment:
Perhaps to avoid passing nulls into the function we could let `BuyerComparator` take in `<?>` but i'm not sure how complicated this might make the code

Reply:
I thought of doing that also but then we would still have to do an instanceof check to see what type comparator it has so that we can decide whether to do getPrice() or getName() or something else so might as well just put all the comparators in Optionals for clarity also